File: prob004/python/prob004.py
import sys
f = open(sys.argv[1],'r')#sys call

#reference:http://stackoverflow.com/questions/16233593/how-to-strip-comma-in-python-string
#remove new line,comma and period. rstrip does not work for commas.
line = f.readline().rstrip('\n.').replace(',','')

wordList = line.split(' ')
pdcTable = {}
for word in wordList:
   if wordList.index(word)+1 in [1, 5, 6, 7, 8, 9, 15, 16, 19]:#required by probstatement
      pdcTable[word[0]]=wordList.index(word)+1
   else:
      pdcTable[word[0]+word[1]]=wordList.index(word)+1

#final output
for key in pdcTable:
   print key, pdcTable[key]

f.close()

File: prob004/ruby/prob004.rb
#problem answer shows up in different prder from python implementation
#because the data structures used were different.

f = File.open(ARGV[0],"r")#sys call to read file
line = f.gets().gsub('.','')#trim .
wordList = line.split(" ")

#initialize an empty hashtable.
#http://ruby-doc.org/core-1.9.3/Hash.html
pdcTbl = Hash.new()

index = 1

#classical for each loop
wordList.each do |word|
   #http://stackoverflow.com/questions/1986386/check-if-value-exists-in-array-in-ruby
   if [1, 5, 6, 7, 8, 9, 15, 16, 19].include? index
      pdcTbl[word[0]]=index
   else
      #http://stackoverflow.com/questions/3525351/how-to-select-array-elements-in-a-given-range-in-ruby
      pdcTbl[word[0..1]]=index 
   end
   index+=1
   #puts word
end

pdcTbl.each_key do |key|
   puts key +" "+ pdcTbl[key].to_s
end
f.close()
